src/image/hfe.c

```diff
@@ -375,11 +380,66 @@ static bool_t hfe_write_track(struct image *im)
             + (im->cur_track & 1) * 256
             + batch_off - im->hfe.write_batch.off
             + (off & 255);
-        for (i = 0; i < nr; i++)
+
+        i = 0;
+
+        if (is_v3 && off == im->hfe.write.start && off > 2) {
```

Comment:
Why not `off >= 2` ?

Reply:
And if the earlier new comment is handled, that could change to `off != 0` or this third element of the conjunction elided altogether!

Reply:
I mean `off==2` is obviously fine to handle, so shouldn't skip that.

You're also going to jog `trk_off==1` in your next round, so `off==1` will never happen here.

You may or may not jog `trk_off==0`. If you do then no test needed as will never see `off==0` here. If you don't then just test `off != 0` needed here.